src/main/kotlin/org/malachite/estella/services/OfferService.kt

```diff
@@ -25,37 +29,67 @@ class OfferService(
+            desiredSkillService
+        ).let {
+            if (!checkAuth(it, jwt).contains(Permission.CREATE)) throw UnauthenticatedException()
+            return this.addOffer(it)
+                .also { recruitmentProcessService.addBasicProcess(it) }
+        }
+    }
 
     fun addOffer(offer: Offer): Offer = offerRepository.save(offer)
 
-    fun updateOffer(id: Int, offer: Offer) {
-        val currOffer: Optional<Offer> = offerRepository.findById(id)
-        if (currOffer.isPresent) {
-            val updated: Offer = currOffer.get().copy(
-                name = offer.name,
-                description = offer.description,
-                position = offer.position,
-                minSalary = offer.minSalary,
-                maxSalary = offer.maxSalary,
-                localization = offer.localization,
-                skills = offer.skills
+    private fun updateOffer(oldOffer: Offer, newOffer: Offer) {
+        val updated: Offer = oldOffer.copy(
+            name = newOffer.name,
+            description = newOffer.description,
+            position = newOffer.position,
+            minSalary = newOffer.minSalary,
+            maxSalary = newOffer.maxSalary,
+            localization = newOffer.localization,
+            skills = newOffer.skills
+        )
+        offerRepository.save(updated)
+    }
+
+    fun updateOffer(id: Int, offerRequest: OfferRequest, jwt: String?) {
+        val currentOffer = this.getOffer(id)
+        if (!checkAuth(currentOffer, jwt).contains(Permission.UPDATE)) throw UnauthenticatedException()
+        this.updateOffer(
+            currentOffer,
+            offerRequest.toOffer(
+                currentOffer.creator,
+                desiredSkillService
             )
-            offerRepository.save(updated)
-        } else {
-            throw OfferNotFoundException()
-        }
+        )
     }
 
-    fun updateOffer(id: Int, offerRequest: OfferRequest, hrPartner: HrPartner) {
-        this.updateOffer(id, offerRequest.toOffer(hrPartner, desiredSkillService))
+    fun deleteOffer(id: Int, jwt: String?) {
+        if (!checkAuth(this.getOffer(id), jwt).contains(Permission.DELETE)) throw UnauthenticatedException()
+        deleteOffer(id)
     }
 
-    fun getHrPartnerOffers(hrPartner: HrPartner): List<OfferResponse> = getOffers()
-            .filter { offer -> offer.creator == hrPartner }
-            .map { offer -> offer.toOfferResponse() }
+    private fun deleteOffer(id: Int) = offerRepository.deleteById(id)
 
-    fun deleteOffer(id: Int) = offerRepository.deleteById(id)
+    private fun checkAuth(offer: Offer, jwt: String?): Set<Permission> {
+        if (securityService.isCorrectApiKey(jwt)) return Permission.allPermissions()
+        val user = securityService.getHrPartnerFromJWT(jwt) ?: securityService.getOrganizationFromJWT(jwt)
+        ?: throw UnauthenticatedException()
+        val permissions = mutableSetOf(Permission.READ)
```

Comment:
To samo co wyżej, ale akurat tutaj chcemy żeby wszystkie oferty można było przeczytać xd